src/segment_path.go

```diff
@@ -220,6 +220,19 @@ func (pt *path) getPwd() string {
 	return pwd
 }
 
+func (pt *path) normalize(inputPath string) string {
```

Comment:
this would need to be put outside of the scope of path when you want to reuse the functionality.

Reply:
Yes, and possibly also refactored a bit.

Reply:
I suppose we leave that refactoring for the following PR?